tutorials/time-freq/plot_ssvep.py

```diff
@@ -0,0 +1,786 @@
+channels.
+"""  # noqa: E501
+# Authors: Dominik Welke <dominik.welke@web.de>
+#          Evgenii Kalenkovich <e.kalenkovich@gmail.com>
+#
+# License: BSD (3-clause)
+
+import matplotlib.pyplot as plt
+import mne
+import numpy as np
+from scipy.stats import ttest_rel
+
+###############################################################################
+# Data preprocessing
+# ------------------
+# Due to a generally high SNR in SSVEP/vSSR, typical preprocessing steps
+# are considered optional. This doesn't mean, that a proper cleaning would not
+# increase your signal quality!
+#
+# * Raw data have FCz reference, so we will apply common-average rereferencing.
+#
+# * We will apply a 50 Hz notch filter to remove line noise,
+#
+# * and a 0.1 - 250 Hz bandpass filter.
+#
+# * Lastly, we will cut the data in 20 s epochs corresponding to the trials.
+#
+#
+
+# Load raw data
+data_path = mne.datasets.ssvep.data_path()
+bids_fname = data_path + '/sub-02/ses-01/eeg/sub-02_ses-01_task-ssvep_eeg.vhdr'
+
+raw = mne.io.read_raw_brainvision(bids_fname, preload=True, verbose=False)
+raw.info['line_freq'] = 50.
+
+# Set montage
+montage_style = 'easycap-M1'
+montage = mne.channels.make_standard_montage(
+    montage_style,
+    head_size=0.095)  # head_size parameter default = 0.095
+raw.set_montage(montage, verbose=False)
+
+# Set common average reference
+raw.set_eeg_reference('average', projection=False, verbose=False)
+
+# Apply notch filter to remove line-noise
+notch = np.arange(raw.info['line_freq'], raw.info['lowpass'] / 2,
+                  raw.info['line_freq'])
+raw.notch_filter(notch, filter_length='auto', phase='zero', verbose=False)
+
+# Apply bandpass filter
+hp = .1
+lp = 250.
+raw.filter(hp, lp, fir_design='firwin', verbose=False)
+
+# Construct epochs
+event_id = {
+    '12hz': 255,
+    '15hz': 155
+}
+events, _ = mne.events_from_annotations(raw, verbose=False)
+raw.info["events"] = events
+tmin, tmax = -1., 20.  # in s
+baseline = None
+epochs = mne.Epochs(
+    raw, events=events,
+    event_id=[event_id['12hz'], event_id['15hz']], tmin=tmin,
+    tmax=tmax, baseline=baseline, verbose=False)
+
+###############################################################################
+# Frequency analysis
+# ------------------
+# Now we compute the frequency spectrum of the EEG data.
+# You will already see the peaks at the stimulation frequencies and some of
+# their harmonics, without any further processing.
+#
+# The 'classical' PSD plot will be compared to a plot of the SNR spectrum.
+# SNR will be computed as a ratio of the power in a given frequency bin
+# to the average power in its neighboring bins.
+# This procedure has two advantages over using the raw PSD:
+#
+# * it normalizes the spectrum and accounts for 1/f power decay.
+#
+# * power modulations which are not very narrow band will disappear.
+#
+#
+
+###############################################################################
```

Comment:
```suggestion
#
```
No need to separate continuous text blocks

Reply:
Fixed.